core/src/main/java/com/linecorp/armeria/client/ClientFactoryBuilder.java

```diff
@@ -560,6 +561,7 @@ private ClientFactoryOptions buildOptions() {
      * Returns a newly-created {@link ClientFactory} based on the properties of this builder.
      */
     public ClientFactory build() {
+        RequestContextUtil.init();
```

Comment:
I'd put this in a `static` block in this class instead of here. Mostly same affect but clearer it's a one-time initialization